pkg/templates/kubeadm/v1beta3/kubeadm.go

```diff
@@ -499,3 +495,20 @@ func kubeProxyConfiguration(s *state.State) *kubeproxyv1alpha1.KubeProxyConfigur
 
 	return kubeProxyConfig
 }
+
+func etcdVersionCorruptCheckExtraArgs(kubeSemVer *semver.Version, etcdImageTag string) (string, map[string]string) {
+	etcdExtraArgs := map[string]string{}
+	if etcdIntegrityCheckConstraint.Check(kubeSemVer) {
+		// This is required because etcd v3.5-[0-2] (used for Kubernetes 1.22+)
+		// has an issue with the data integrity.
+		// See https://groups.google.com/a/kubernetes.io/g/dev/c/B7gJs88XtQc/m/rSgNOzV2BwAJ
+		// for more details.
+		if etcdImageTag == "" {
+			etcdImageTag = "3.5.3-0"
+		}
+		etcdExtraArgs["experimental-initial-corrupt-check"] = "true"
+		etcdExtraArgs["experimental-corrupt-check-time"] = "240m"
+	}
+
+	return etcdImageTag, etcdExtraArgs
+}
```

Comment:
This was required because linter was failing:

```
pkg/templates/kubeadm/v1beta3/kubeadm.go:60:1: cyclomatic complexity 31 of func `NewConfig` is high (> 30) (gocyclo)
func NewConfig(s *state.State, host kubeoneapi.HostConfig) ([]runtime.Object, error) {
```